code/modules/mob/living/carbon/human/life.dm

```diff
@@ -439,13 +439,42 @@
 		var/failed_inhale = 0
 		var/failed_exhale = 0
 
-		if(species.has_organ["breathing apparatus"])
+		if(species.has_organ["breathing apparatus"] && src.get_species() == "Vaurca")
 			var/obj/item/organ/vaurca/breathingapparatus/L = internal_organs_by_name["breathing apparatus"]
 			if(isnull(L))
 				poison_type = null
 			else if(L.is_broken())
 				poison_type = "oxygen" //if Vaurca breathing apparatus breaks, oxygen becomes poisonous.
 
+		if(species.has_organ["neural socket"])
+			var/obj/item/organ/vaurca/neuralsocket/N = internal_organs_by_name["neural socket"]
+
+			if(!isnull(N) && !N.is_broken() && !(all_languages["Hivenet"] in src.languages))
+				add_language("Hivenet")
+				src << "\blue Your mind expands, and your thoughts join the unity of the Hivenet."
+
+			else
+				if(isnull(N) && all_languages["Hivenet"] in src.languages)
+					remove_language("Hivenet")
+					src << "\red Your mind suddenly grows dark as the unity of the Hive is torn from you."
+
+				if(N.is_broken() && all_languages["Hivenet"] in src.languages)
+					remove_language("Hivenet")
+					src << "\red Your mind suddenly grows dark as the unity of the Hive is torn from you."
+
+		if(l_hand && istype(l_hand, /obj/item/weapon/radio/vaurca) && !(src.get_species() == "Vaurca"))

```

Comment:
I'm dubious of this. Is there any better way to manage the assignment of this language, specially in cases of languages?

Perhaps look at how binary speech works with the binary headset/implant.

Also, the if/elseif/else structure here looks garbled. The else condition on line 473 only applies to the if on line 469. Ergo, it removes the language even if your left hand has a radio.